src/WebJobs.Script.WebHost/Metrics/LinuxContainerMetricsPublisher.cs

```diff
@@ -0,0 +1,227 @@
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost.Metrics
+{
+    public class LinuxContainerMetricsPublisher
+    {
+        public const string PublishMemoryActivityPath = "/memoryactivity";
+        public const string PublishFunctionActivityPath = "/functionactivity";
+
+        public const string ContainerNameHeader = "x-ms-functions-container-name";
+        public const string HostNameHeader = "x-ms-functions-host-name";
+        public const string SiteTokenHeader = "x-ms-site-restricted-token";
+        public const string StampNameHeader = "x-ms-site-home-stamp";
+
+        private const string _requestUriFormat = "https://{0}:31002/api/metrics";
+        private const int _maxErrorLimit = 5;
+        private const int _maxBufferSize = 50000;
+
+        private readonly TimeSpan _memorySnapshotInterval = TimeSpan.FromMilliseconds(1000);
+        private readonly TimeSpan _metricPublishInterval = TimeSpan.FromMilliseconds(30 * 1000);
+        private readonly IOptionsMonitor<StandbyOptions> _standbyOptions;
+        private readonly IDisposable _standbyOptionsOnChangeSubscription;
+        private readonly string _requestUri;
+        private readonly IEnvironment _environment;
+
+        private string _containerName;
+        private Timer _metricsPublisherTimer;
+        private BlockingCollection<MemoryActivity> _memoryActivities;
+        private BlockingCollection<FunctionActivity> _functionActivities;
+
+        private ConcurrentQueue<MemoryActivity> _currentMemoryActivities;
+        private ConcurrentQueue<FunctionActivity> _currentFunctionActivities;
+
+        private HttpClient _httpClient;
+        private Action<LogLevel, string> _logTraceEvent;
+        private Process _process;
+        private Timer _processMonitorTimer;
+        private int _errorCount = 0;
+        private string _hostName;
+        private string _stampName;
+
+        public LinuxContainerMetricsPublisher(IEnvironment environment, IOptionsMonitor<StandbyOptions> standbyOptions, Action<LogLevel, string> logMetricsPublishEvent, HttpClient httpClient)
+        {
+            _httpClient = httpClient ?? throw new ArgumentNullException(nameof(httpClient));
+            _standbyOptions = standbyOptions ?? throw new ArgumentNullException(nameof(standbyOptions));
+            _environment = environment ?? throw new ArgumentNullException(nameof(environment));
+            _logTraceEvent = logMetricsPublishEvent ?? throw new ArgumentNullException(nameof(logMetricsPublishEvent));
+            _memoryActivities = new BlockingCollection<MemoryActivity>(new ConcurrentQueue<MemoryActivity>(), boundedCapacity: _maxBufferSize);
+            _functionActivities = new BlockingCollection<FunctionActivity>(new ConcurrentQueue<FunctionActivity>(), boundedCapacity: _maxBufferSize);
+
+            _currentMemoryActivities = new ConcurrentQueue<MemoryActivity>();
+            _currentFunctionActivities = new ConcurrentQueue<FunctionActivity>();
```

Comment:
What's the difference between these pairs of collections? Can you please add comments to make their roles clear?